charts/kyverno/templates/admission-controller/deployment.yaml

```diff
@@ -184,6 +184,10 @@ spec:
             {{- if not .Values.admissionController.metering.disabled }}
             - --otelConfig={{ .Values.admissionController.metering.config }}
             - --metricsPort={{ .Values.admissionController.metering.port }}
+            {{- if .Values.admissionController.metering.secure }}
+            - --metricsCaSecretName={{ template "kyverno.admission-controller.name" . }}.{{ template "kyverno.namespace" . }}.metering.kyverno-tls-ca
```

Comment:
You setting both flags with a static secret name that is only created when `createSelfSignedCert` is set to true, which is not checked. So you enforce a defined secret name and do not allow a custom secret (name). Is this expected?